server/src/main/java/org/opensearch/common/cache/store/listeners/StoreAwareCacheEventListener.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.store.listeners;
+
+import org.opensearch.common.cache.store.StoreAwareCacheRemovalNotification;
+import org.opensearch.common.cache.store.enums.CacheStoreType;
+
+/**
+ * This can be used to listen to tiered caching events
+ * @param <K> Type of key
+ * @param <V> Type of value
+ *
+ * @opensearch.internal
+ */
+public interface StoreAwareCacheEventListener<K, V> {
+
+    void onMiss(K key, CacheStoreType cacheStoreType);
+
+    void onRemoval(StoreAwareCacheRemovalNotification<K, V> notification);
+
+    void onHit(K key, V value, CacheStoreType cacheStoreType);
+
+    void onCached(K key, V value, CacheStoreType cacheStoreType);
```

Comment:
The `CacheStoreType` as a second argument hasn't found any use so far, not sure if this is needed in future as well since the Cache would already encapsulate the specific type of listener and there isn't a consumer In could find that would rely on the store type

Reply:
This listener will be eventually implemented by IndicesRequestCache(consumer) and then store type would be used to calculate stats(hits/missed/cached) accordingly for desired tier.

I will raise a follow up PR to integrate onHeap/tiered cache which will make things cleared.

Reply:
Cool, will discuss this over the next PR